File: L002_Basics/fibTillN.cpp
#include <iostream>
using namespace std;

void fibTillN(int n){
    int a = 0 , b = 1;
    if (n < 1 )
        return;
    cout<<a<<endl;
    
    for(int i=1;i<n;i++){
        cout<<b<<endl;
        int c = a+b;
        a = b;
        b = c;
    }
}

int main()
{
    int n;
    cin >> n;
    
    fibTillN(n);
    
}

File: L002_Basics/fibTillN.java
import java.util.Scanner;

public class fibTillN {

    public static Scanner sc = new Scanner(System.in);

    public static void fibTillN_(int n) {
        int a = 0, b = 1;
        if (n < 1)
            return;
        System.out.println(a + " ");

        for (int i = 1; i < n; i++) {
            System.out.println(b + " ");
            int c = a + b;
            a = b;
            b = c;
        }
    }

    public static void main(String[] args) {

        int n = sc.nextInt();
        fibTillN_(n);

    }
}